app/models/concerns/roleable.rb

```diff
@@ -18,6 +18,16 @@ def regular?
     !staff?
   end
 
+  def whisperer?
+    @whisperer ||= begin
+      return false if !SiteSetting.enable_whispers?
+      return true if staff?
+      return false if SiteSetting.whispers_allowed_groups.blank?
+      return true if SiteSetting.whispers_allowed_group_ids.include?(primary_group_id)
+      group_users&.exists?(group_id: SiteSetting.whispers_allowed_group_ids)
```

Comment:
```suggestion
      whispers_allowed_group_ids = SiteSetting.whispers_allowed_group_ids
      return false if whispers_allowed_group_ids.blank?
      return true if whispers_allowed_group_ids.include?(primary_group_id)
      group_users&.exists?(group_id: whispers_allowed_group_ids)
```

The return value of the site setting is not memoized so let's allocate an array each time the method is called.